### PR TITLE
Import Custom Field as Select Options in Sage Intacct

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts [hunk 1]
```diff
@@ -131,6 +134,21 @@ export class ConfigurationComponent implements OnInit {
 
       that.generalSettingsForm.controls.reimburExpense.disable();
 
+
+      const fyleProjectMapping = that.mappingSettings.filter(
+        setting => setting.source_field === 'PROJECT' && setting.destination_field !== 'PROJECT'
+      );
+
+      const sageIntacctProjectMapping = that.mappingSettings.filter(
```
Comment (NileshPant1999) [inline src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts:142]: need to remove this

File: src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts [hunk 2]
```diff
@@ -120,13 +132,26 @@ export class ConfigurationComponent implements OnInit {
       that.generalSettingsForm = that.formBuilder.group({
         reimburExpense: [that.generalSettings ? that.generalSettings.reimbursable_expenses_object : ''],
         cccExpense: [that.generalSettings ? that.generalSettings.corporate_credit_card_expenses_object : ''],
-        importProjects: [that.generalSettings.import_projects],
+        importProjects: [false],
```
Comment (NileshPant1999) [inline src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts:135]: importProjects

File: src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts [hunk 3]
```diff
@@ -89,16 +90,25 @@ export class ConfigurationComponent implements OnInit {
 
   getAllSettings() {
     const that = this;
-    that.isLoading = true;
+
     forkJoin(
       [
         that.settingsService.getGeneralSettings(that.workspaceId),
-        that.settingsService.getMappingSettings(that.workspaceId)
+        that.settingsService.getMappingSettings(that.workspaceId),
```
Comment (Sravanksk) [inline src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts:97]: remove `,`

**Review by Sravanksk — APPROVED** [review]

File: src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts [hunk 4]
```diff
@@ -99,6 +100,15 @@ export class ConfigurationComponent implements OnInit {
       that.generalSettings = responses[0];
       that.mappingSettings = responses[1].results;
 
+      const projectFieldMapping = that.mappingSettings.filter(
+        setting => (setting.source_field === 'PROJECT' && setting.destination_field === 'PROJECT')
```
Comment (ashwin1111) [inline src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts:104]: check netsuite

File: src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts [hunk 5]
```diff
@@ -120,13 +130,26 @@ export class ConfigurationComponent implements OnInit {
       that.generalSettingsForm = that.formBuilder.group({
         reimburExpense: [that.generalSettings ? that.generalSettings.reimbursable_expenses_object : ''],
         cccExpense: [that.generalSettings ? that.generalSettings.corporate_credit_card_expenses_object : ''],
-        importProjects: [that.generalSettings.import_projects],
+        importProjects: [importProjects],
```
Comment (ashwin1111) [inline src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts:133]: check netsuite, we send as false every time

File: src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts [hunk 6]
```diff
@@ -120,13 +130,26 @@ export class ConfigurationComponent implements OnInit {
...
 
+      const fyleProjectMapping = that.mappingSettings.filter(
+        setting => setting.source_field === 'PROJECT' && setting.destination_field !== 'PROJECT'
+      );
+
+      const sageIntacctProjectMapping = that.mappingSettings.filter(
+        setting => setting.destination_field === 'PROJECT' && setting.source_field !== 'PROJECT'
+      );
+
+      // disable project sync toggle if either of Fyle / SageIntacct Projects are already mapped to different fields
```
Comment (ashwin1111) [inline src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts:148]: Sage Intacct

File: src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts [hunk 7]
```diff
@@ -180,15 +204,15 @@ export class ConfigurationComponent implements OnInit {
       const cccExpensesObject = that.generalSettingsForm.value.cccExpense || (that.generalSettings ? that.generalSettings.corporate_credit_card_expenses_object : null);
       const categoryMappingObject = that.getCategory(reimbursableExpensesObject)[0].value;
       const employeeMappingsObject = that.getEmployee(reimbursableExpensesObject)[0].value;
-      const importProjects = that.generalSettingsForm.value.importProjects;
+      const importProjects = that.generalSettingsForm.value.importProjects ? that.generalSettingsForm.value.importProjects : false;
```
Comment (ashwin1111) [inline src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts:207]: check netsuite

File: src/app/si/settings/sage-intacct-configurations/expense-field-configuration/expense-field-configuration.component.html [hunk 8]
```diff
@@ -1,59 +1,138 @@
...
+            </mat-form-field>
+
+            <p *ngIf="isSystemField" class="existing-field-warning">
+              <mat-icon [inline]="true" class="info-icon warning">info</mat-icon>
+              This field already exists. Please enter new.
+            </p>
+
+            <div class="custom-field-grp" fxLayoutAlign="start stretch">
+              <div fxFlex="10"></div>
+              <a fxLayout="row" fxLayoutAlign="start center" class="custom-field-btn" mat-raised-button
```
Comment (ashwin1111) [inline src/app/si/settings/sage-intacct-configurations/expense-field-configuration/expense-field-configuration.component.html:111]: We agreed not to use <a> tag right?

Comment (NileshPant1999) [reply src/app/si/settings/sage-intacct-configurations/expense-field-configuration/expense-field-configuration.component.html:111]: yes removing this off

**Pull request merged**